### PR TITLE
External Organ Rework: new bodypart_overlay system

### LINKED ISSUE #71820: Flight Potion does not work for Moths
Reporting client version: 514.1588

## Round ID:
[195847](https://scrubby.melonmesa.com/round/195847)

## Testmerges:
- [Partially reverts moving up/down whilst in a movable from #70431](https://www.github.com/tgstation/tgstation/pull/71527)

## Reproduction:

Moth gets strange elixir that should grant large wings for the purpose of flight. Moth drinks potion. Moth gets text saying " A terrible pain travels down your back as your wings change shape!" Moth gets no large wings, also experiences no pain, so it could be worse.

### PR DESCRIPTION
**Pull request by Time-Green**
Bodypart overlays are now drawn by the new /datum/bodypart_overlay datum. 

External organs no longer draw anything and instead add a special /datum/bodypart_overlay/mutant to the bodypart, which draws everything

Makes it way easier to add custom overlays to limbs, since the whole system is now modularized and external organs are just one implementation of it

I haven't moved anything but external organs to this new system, I'll move eyes, bodymarkings, hair, lipstick etc to this later

New pipeline is as follows:
- External organ added to limb
- External organ adds /datum/bodypart_overlay/mutant to limb to bodypart_overlays
- Limb updates its icon, looks for all /datum/bodypart_overlay in bodypart_overlays
- Very cool new overlay on your limb!

closes #71820

:cl:
refactor: External organs have been near-completely refactored. 
admin: Admin-spawned external organs will load with a random icon and color
fix: fixes angel wings not working for non-humans (it was so fucking broken)
fix: fixes external organs being invisible if they werent initialized with a human
/:cl:

### Why this is good for the game
External organs are cool but are pretty limited in some ways. Making stuff like synthetic organs is kinda fucked. I tried and it was dogshit. Now you can just give an icon state and icon and you're good (using /datum/bodypart_accessory/simple)

Stuff like eyes, cat ears and hair seem like good choices for extorgans, but don't quite work for it because their icons work a lot differently. This solves for it completely since any organ (or object or whatever) can add it's own icon to a bodypart. 

Want to add an iron plate to someones head? Go ahead. Want a heart to stick out of someones chest? No problem.

### CODE REVIEW

**Comment by LemonInTheDark** [comment]
is organ the right word for this?

**Comment by Time-Green** [comment]
> is organ the right word for this?

External organs is definitely not correct terminology but I've grown attached to it

File: code/datums/dna.dm [hunk 1]
```diff
@@ -613,7 +613,7 @@ GLOBAL_LIST_INIT(total_uf_len_by_block, populate_total_uf_len_by_block())
 	if(dna.features["pod_hair"])
 		dna.features["pod_hair"] = GLOB.pod_hair_list[deconstruct_block(get_uni_feature_block(features, DNA_POD_HAIR_BLOCK), GLOB.pod_hair_list.len)]
 
-	for(var/obj/item/organ/external/external_organ as anything in external_organs)
+	for(var/obj/item/organ/external/external_organ in internal_organs)
```
Comment (Kapu1178) [inline code/datums/dna.dm:616]: bro :sob:

**Comment by lizardqueenlexi** [comment]
i thought calling things related to non-humans "mutant" wasn't really much of a thing anymore (except for some old stuff that's still lying around)

File: code/datums/mood_events/generic_negative_events.dm [hunk 2]
```diff
@@ -335,11 +335,6 @@
 	mood_change = -2
 	timeout = 5 MINUTES
 
-/datum/mood_event/tail_regained_wrong
```
Comment (optimumtact) [inline code/datums/mood_events/generic_negative_events.dm:338]: why did you remove this event is it unused?

Comment (Kapu1178) [reply code/datums/mood_events/generic_negative_events.dm:338]: `del: removes negative moodevent on getting wrong species tail`

`I also removed the mood-event on getting the wrong tail. It was turbo shitcode and causing so many hard-dels for someting that discourages frankensteining???`

Comment (optimumtact) [reply code/datums/mood_events/generic_negative_events.dm:338]: add it back

**Comment by optimumtact** [comment]
also what value does the external organ have if the bodypart overlay is doing all the work? is it primarily to interact with the surgery system?

File: code/datums/mood_events/generic_negative_events.dm (hunk 2 above)
Comment (Kapu1178) [reply code/datums/mood_events/generic_negative_events.dm:338]: @Time-Green ^

Comment (Time-Green) [reply code/datums/mood_events/generic_negative_events.dm:338]: AAAARGH

File: code/datums/dna.dm (hunk 1 above)
Comment (Time-Green) [reply code/datums/dna.dm:616]: External organs are now in internal organs yes, it’s a lot better than having double definitions everywhere. Maybe I’ll have to rename it at one point

**Comment by Time-Green** [comment]
> also what value does the external organ have if the bodypart overlay is doing all the work? is it primarily to interact with the surgery system?

Yeah external organs just interact with the surgery system to make them transferable, but otherwise they don’t do much anymore and it’s all handled bu the new datum

File: code/modules/unit_tests/create_and_destroy.dm [hunk 3]
```diff
@@ -30,6 +30,8 @@ GLOBAL_VAR_INIT(running_create_and_destroy, FALSE)
 		/obj/merge_conflict_marker,
 		//briefcase launchpads erroring
 		/obj/machinery/launchpad/briefcase,
+		//Both are abstract types meant to scream bloody murder if spawned in raw
+		/obj/item/organ/external, /obj/item/organ/external/wings,
```
Comment (Mothblocks) [inline code/modules/unit_tests/create_and_destroy.dm:34]:
```suggestion
		/obj/item/organ/external,
		/obj/item/organ/external/wings,
```

i know why you did this but i dont agree

**Comment by Time-Green** [comment]
Ran into a small bug I dont have time to fix, will be ready again in a few hours

File: code/datums/dna.dm (hunk 1 above)
Comment (tralezab) [inline code/datums/dna.dm:616]: 100% in the future `internal organs` has to be renamed but for the sake of keeping prs reviewable and moving i'll let it slide

**Review by tralezab — APPROVED** [review]

File: code/datums/dna.dm (hunk 1 above)
Comment (Time-Green) [reply code/datums/dna.dm:616]: I'll do it tomorrow

Comment (Time-Green) [reply code/datums/dna.dm:616]: no wait Ill do it after fikou finishes his organ species rework
